src/ActionList/List.tsx

```diff
@@ -0,0 +1,69 @@
+import {Group, GroupProps} from './Group'
+import {Item, ItemProps} from './Item'
+import React from 'react'
+import {Divider} from './Divider'
+import {Header, HeaderProps} from './Header'
+import styled from 'styled-components'
+import {get} from '../constants'
+
+type Flatten<T extends unknown> = T extends (infer U)[] ? U : never
+
+interface UngroupedListProps {
+  items: ItemProps[]
+  renderItem?: (props: ItemProps) => JSX.Element
+}
+function isUngroupedListProps(props: ListProps): props is UngroupedListProps {
+  return typeof props === 'object' && props !== null && !('groupMetadata' in props)
```

Comment:
I believe the first two checks are always true.

```suggestion
  return !('groupMetadata' in props)
```

Reply:
Removed in https://github.com/primer/components/pull/1078/commits/d92d5552068368742de614aef522e564e88c4226!